### PR TITLE
Respect opt-outs from dotnet/msbuild#7008

### PR DESCRIPTION
**Pull request by Forgind**
See https://github.com/dotnet/msbuild/pull/7008 for details.

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props
Comment (rainersigwald) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props:77]:
```suggestion
    <!-- By default exclude GAC, registry, output directory from search paths. -->
```

**Review by rainersigwald — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props
Comment (Nirmal4G) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props:81]:
```suggestion
    <AssemblySearchPaths Condition="'$(AssemblySearchPath_UseRawFileName)' != 'false'">$(AssemblySearchPaths);{RawFileName}</AssemblySearchPaths>
```

**CHANGES**

- Use quotes to avoid erroring out with a malformed value (_with space and illegal symbols_) entering the condition.
- Wouldn't it better to use `BaseProperty_SubOption` with an `_` acting as a separator to make these properties clearer?

See that we already use this pattern of naming here...

https://github.com/dotnet/msbuild/blob/14313d17e407e24ad80e113bd475d1dd7c698656/src/Tasks/Microsoft.Common.props#L181

Comment (Forgind) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props:81]:
```suggestion
    <AssemblySearchPaths Condition="'$(AssemblySearchPathUseCandidateAssemblyFiles)' != 'false'">{CandidateAssemblyFiles}</AssemblySearchPaths>
    <AssemblySearchPaths Condition="'$(AssemblySearchPathUseHintPathFromItem)' != 'false'">$(AssemblySearchPaths);{HintPathFromItem}</AssemblySearchPaths>
    <AssemblySearchPaths Condition="'$(AssemblySearchPathUseTargetFrameworkDirectory)' != 'false'">$(AssemblySearchPaths);{TargetFrameworkDirectory}</AssemblySearchPaths>
    <AssemblySearchPaths Condition="'$(AssemblySearchPathUseRawFileName)' != 'false'">$(AssemblySearchPaths);{RawFileName}</AssemblySearchPaths>
```

Comment (Forgind) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props]: I added the quotes. I think your BaseProperty_SubOption suggestion probably would've been a bit clearer, but I'm not super interested in changing the code we merged ~4 weeks ago, so I'm voting to leave it as-is on that point.

Comment (Nirmal4G) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props]: This change is not yet shipped with MSBuild, right? Meaning, it's not too late to change it. I could open a PR there if it's okay with you?

Comment (Forgind) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props]: I'm fine with that. You're right that we haven't shipped it yet, but a lot of people are off right now for the holidays. I won't promise it'll get in, but if you make it, I'll approve, and if one more person approves early enough, we can merge it. Sound reasonable?

Make sure to update the docs, too.

Comment (Nirmal4G) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props]: I have made the changes. Here's the PR dotnet/msbuild#7166.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props
Comment (Nirmal4G) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props:81]:
```suggestion
    <AssemblySearchPaths Condition="'$(AssemblySearchPath_UseCandidateAssemblyFiles)' != 'false'">{CandidateAssemblyFiles}</AssemblySearchPaths>
    <AssemblySearchPaths Condition="'$(AssemblySearchPath_UseHintPathFromItem)' != 'false'">$(AssemblySearchPaths);{HintPathFromItem}</AssemblySearchPaths>
    <AssemblySearchPaths Condition="'$(AssemblySearchPath_UseTargetFrameworkDirectory)' != 'false'">$(AssemblySearchPaths);{TargetFrameworkDirectory}</AssemblySearchPaths>
    <AssemblySearchPaths Condition="'$(AssemblySearchPath_UseRawFileName)' != 'false'">$(AssemblySearchPaths);{RawFileName}</AssemblySearchPaths>
```

**Review by Nirmal4G — CHANGES_REQUESTED** [review]
Since the PR is now merged, we can update the property names.

**Comment by Forgind** [comment]
> Since the PR is now merged, we can update the property names.

I think we have to wait until the change has flowed into the SDK, but that should happen at some point, and we shouldn't merge this until then anyway.

**Review by Nirmal4G — APPROVED** [review]
Now it's looking good! 😏

**Pull request merged**
